### PR TITLE
Migrate `run-make/allow-non-lint-warnings-cmdline` to `rmake.rs`

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of https://github.com/rust-lang/rust/issues/121876.

r? @jieyouxu

### CODE REVIEW

File: tests/run-make/allow-non-lint-warnings-cmdline/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Test that -A warnings makes the 'empty trait list for derive' warning go away.
```
Comment (jieyouxu) [inline tests/run-make/allow-non-lint-warnings-cmdline/rmake.rs:1]: Discussion: I thought about this a bit, could this be written as a UI test? Something like

```rs
//@ compile-flags: -Awarnings

#[derive()]
#[derive(Copy, Clone)]
pub struct Foo;

pub fn main() {}
```

and then obviously the stderr would be blessed which would not have any warnings?

I'm trying this locally, let me get back to you.

**Comment by GuillaumeGomez** [comment]
And fixed the formatting issue.

File: tests/run-make/allow-non-lint-warnings-cmdline/rmake.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply tests/run-make/allow-non-lint-warnings-cmdline/rmake.rs:1]: Yes it could actually. Do you prefer me to switch this test to a ui test?

Comment (jieyouxu) [reply tests/run-make/allow-non-lint-warnings-cmdline/rmake.rs:1]: Yeah, if it doesn't really need the facilities of run-make tests, it's usually better as ui tests.

**Comment by GuillaumeGomez** [comment]
Turned it into a ui test. :)

**Review by jieyouxu — APPROVED** [review]
Thanks, r=me after CI is green

**Pull request merged**